src/jvmTest/java/it/krzeminski/snakeyaml/engine/kmp/internal/FSPath.kt

```diff
@@ -0,0 +1,46 @@
+package it.krzeminski.snakeyaml.engine.kmp.internal
+
+import okio.FileSystem
+import okio.Path
+import okio.Path.Companion.toPath
+import okio.Source
+
+/**
+ * Represents a [FileSystem]-aware [Path]. Thanks to this, there's no need
+ * to keep writing e.g. `FileSystem.SYSTEM.exists("foobar".toPath())`, and
+ * repeat the "file system" part. The file system becomes a part of the path,
+ * allowing more concise and abstract code - specifying the file system is
+ * required only once, at the beginning of operating on the path.
+ */
+class FSPath(val path: Path, val fs: FileSystem) {
```

Comment:
note: Not sure, but looks like those properties can be private

Reply:
It's true that nothing uses them right now, but I think having access to these can be handy. We're not publishing a library so it's not crucial to think about it right now - I'd opt for flexibility, especially in the tests.